packages/common/src/utils/fields.ts

```diff
@@ -44,15 +41,19 @@ export const getFieldsFromMetricQuery = (
         }
         return acc;
     }, {});
-    const additionalMetrics = metricQuery.additionalMetrics?.reduce<
-        Record<string, AdditionalMetric>
-    >(
-        (acc, additionalMetric) => ({
-            ...acc,
-            [getItemId(additionalMetric)]: additionalMetric,
-        }),
-        {},
-    );
+    const additionalMetrics = (metricQuery.additionalMetrics || []).reduce<
+        Record<string, Metric>
+    >((acc, additionalMetric) => {
+        const table = explore.tables[additionalMetric.table];
+        if (table) {
+            const metric = convertAdditionalMetric({
```

Comment:
why do we want to convert ? is this what the explore does ?

Reply:
yes, the VizProvider converts it atm.